src/pages/rss.xml.ts

```diff
@@ -0,0 +1,100 @@
+// A fast, cached build-time only RSS.
+// Currently returns title, pubDate, link, guid & description.
+// Description is truncated to 200 chars.
+// Image support exists, just needed to figure out the fuckery of XML.
+
+import rss from "@astrojs/rss";
+import type { APIRoute } from "astro";
+import { getCollection, type CollectionEntry } from "astro:content";
+
+const baseUrl = 'https://nn1.dev';
+
+let cachedXML: string | null = null;
+type SpotlightEntry = CollectionEntry<"spotlight">;
+
+export const GET: APIRoute = async () => {
+  if (cachedXML) {
+    return new Response(cachedXML, {
+      headers: { "Content-Type": "application/rss+xml; charset=utf-8" },
+    });
+  }
+  try {
+    const now = new Date();
+    const spotlights: SpotlightEntry[] = await getCollection("spotlight");
+
+    // Filter out future-dated posts
+    const publishedSpotlights = spotlights.filter(spotlight =>
```

Comment:
We can simplify this one by filtering out all of the spotlights that are in the draft mode.

```
spotlights.filter(s => !s.draft)
```

Something like that should do.